### PR TITLE
fixes #11240 - Add api docs to hostgroup params

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

**Comment by shlomizadok** [comment]
oops, I did it again...

**Comment by domcleal** [comment]
Please use "fixes" instead of "refs".

File: app/controllers/api/v2/hostgroups_controller.rb [hunk 1]
```diff
@@ -27,18 +27,19 @@ def show
...
-          param :operatingsystem_id, :number
-          param :architecture_id, :number
-          param :medium_id, :number
-          param :ptable_id, :number
-          param :puppet_ca_proxy_id, :number
-          param :subnet_id, :number
-          param :domain_id, :number
-          param :realm_id, :number
-          param :puppet_proxy_id, :number
+          param :name, String, :required => true, :desc => N_('Name of the hostgroup')
```
Comment (domcleal) [inline app/controllers/api/v2/hostgroups_controller.rb]: Use "host group" throughout please.

File: app/controllers/api/v2/hostgroups_controller.rb [hunk 2]
```diff
@@ -27,18 +27,19 @@ def show
...
+          param :parent_id, :number, :desc => N_('Parent id of the hostgroup')
```
Comment (domcleal) [inline app/controllers/api/v2/hostgroups_controller.rb]: Use "ID" (capitalised) throughout please.

File: app/controllers/api/v2/hostgroups_controller.rb [hunk 3]
```diff
@@ -27,18 +27,19 @@ def show
...
+          param :environment_id, :number, :desc => N_("hostgroup's environment id")
```
Comment (domcleal) [inline app/controllers/api/v2/hostgroups_controller.rb]: I don't think you need "hostgroup's" on every parameter!

**Comment by shlomizadok** [comment]
@domcleal - ammended all your comments

File: app/controllers/api/v2/hostgroups_controller.rb [hunk 4]
```diff
@@ -27,18 +27,19 @@ def show
...
+          param :parent_id, :number, :desc => N_('Parent ID of the host group')
+          param :environment_id, :number, :desc => N_('Environment ID')
+          param :operatingsystem_id, :number, :desc => N_('Operating system ID')
+          param :architecture_id, :number, :desc => N_('Architecture id')
+          param :medium_id, :number, :desc => N_('Media ID')
+          param :ptable_id, :number, :desc => N_('Partition table ID')
+          param :puppet_ca_proxy_id, :number, :desc => N_('Puppet CA proxy ID')
+          param :subnet_id, :number, :desc => N_('Subnet ID')
+          param :domain_id, :number, :desc => N_('Domain ID')
+          param :realm_id, :number, :desc => N_('Realm ID, if exists')
```
Comment (domcleal) [inline app/controllers/api/v2/hostgroups_controller.rb]: "Realm proxy ID" (it's a smart proxy again), not sure what the "if exists" is for?

File: app/controllers/api/v2/hostgroups_controller.rb [hunk 5]
```diff
@@ -27,18 +27,19 @@ def show
...
+          param :puppet_proxy_id, :number, :desc => N_('Puppet proxy ID')
+          param :root_pass, String, :desc => N_('Operatingsystem root password')
```
Comment (domcleal) [inline app/controllers/api/v2/hostgroups_controller.rb]: "Operating system", or perhaps something more like the root_pass setting, e.g. "Root password on provisioned hosts"?

**Comment by shlomizadok** [comment]
ammended. Thanks, @domcleal

File: app/controllers/api/v2/hostgroups_controller.rb [hunk 6]
```diff
@@ -27,18 +27,19 @@ def show
...
-          param :puppet_ca_proxy_id, :number
-          param :subnet_id, :number
-          param :domain_id, :number
-          param :realm_id, :number
-          param :puppet_proxy_id, :number
+          param :name, String, :required => true, :desc => N_('Name of the host group')
+          param :parent_id, :number, :desc => N_('Parent ID of the host group')
+          param :environment_id, :number, :desc => N_('Environment ID')
+          param :operatingsystem_id, :number, :desc => N_('Operating system ID')
+          param :architecture_id, :number, :desc => N_('Architecture id')
```
Comment (dLobatog) [inline app/controllers/api/v2/hostgroups_controller.rb:34]: missed this 'id' -> ID, I'll change it on merge.

**Comment by dLobatog** [comment]
Merged as 8caf49dbe303f791b8bde3065aae535601d959f0, thanks @shlomizadok !

**Pull request closed without merging**